### PR TITLE
New Document entity object to encapsulate elasticsearch result

### PR DESCRIPTION
**Pull request by theus77**

### CODE REVIEW

File: Entity/Document.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private $source;
+
+    public function __construct(string $contentType, string $ouuid, array $source)
+    {
+        $this->contentType = $contentType;
+        $this->ouuid = $ouuid;
+        $this->source = $source;
+    }
+
+    /**
```
Comment (dameert) [inline Entity/Document.php]: I would remove redundant docblocks, all information of this doc is already readable in code.

File: Entity/Document.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public function getContentType(): string
+    {
+        return $this->contentType;
+    }
+
+    /**
+     * @param string $contentType
+     * @return Document
+     */
+    public function setContentType(string $contentType): Document
```
Comment (dameert) [inline Entity/Document.php]: remove the setters. You should not be allowed to modify a Document that comes from a search operation. That would alter the reality. 

If one really needs a modification in a Document, they should create a new Document with the values of the old one that they want to keep, and pass the modification via the constructor

File: Entity/Document.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     * @param array $source
+     * @return Document
+     */
+    public function setSource(array $source): Document
+    {
+        $this->source = $source;
+        return $this;
+    }
+
+
```
Comment (dameert) [inline Entity/Document.php]: I think @Davidmattei  wanted to add a __get function that will look for the key in the _source of the document. Can you confirm this @Davidmattei ?

File: Entity/Document.php (hunk 1 above)
Comment (theus77) [reply Entity/Document.php]: Old habit (auto generated by PHP storm)

File: Entity/Document.php (hunk 2 above)
Comment (theus77) [reply Entity/Document.php]: Agree

File: Entity/Document.php (hunk 3 above)
Comment (theus77) [reply Entity/Document.php]: I guess too, but that's for a futur merge request

Comment (dameert) [reply Entity/Document.php]: ok

**Comment by dameert** [comment]
need to run phpcs

File: Entity/Document.php (hunk 2 above)
Comment (theus77) [reply Entity/Document.php]: Done

**Pull request merged**